gateway/handler_unixfs__redirects.go

```
@@ -101,8 +116,23 @@ func (i *handler) handleRedirectsFileRules(w http.ResponseWriter, r *http.Reques
 			// Or 4xx
 			if rule.Status == 404 || rule.Status == 410 || rule.Status == 451 {
 				toPath := rootPath + rule.To
-				content4xxPath := ipath.New(toPath)
-				err := i.serve4xx(w, r, content4xxPath, rule.Status)
+				imContent4xxPath, err := NewImmutablePath(ipath.New(toPath))
+				if err != nil {
+					// TODO: this shouldn't happen should we just log? Is this error right?
```

We don't want to swallow error that suggests invalid rule in the file, return err
```suggestion
```